data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -301,17 +319,28 @@ public Map<String, String> getTracingAttributes() {
         return attributes;
     }
 
+    /**
+     * Gets the amount of time a query may execute on the service before it times out, formatted as a KQL timespan.
+     * @param timeoutObj amount of time before timeout, which may be a Long, String or Integer.
+     *                    Value must be between 1 minute and 1 hour, and so value below the minimum or above the maximum will be adjusted accordingly.
+     */
+    // TODO: Breaking change for next major release: static String getTimeoutAsTimespan(Object timeoutObj)
     String getTimeoutAsString(Object timeoutObj) {
-        String timeoutString = "";
-        if (timeoutObj instanceof Long) {
-            Duration duration = Duration.ofMillis((Long) timeoutObj);
-            timeoutString = Utils.formatDurationAsTimespan(duration);
-        } else if (timeoutObj instanceof String) {
-            timeoutString = (String) timeoutObj;
-        } else if (timeoutObj instanceof Integer) {
-            Duration duration = Duration.ofMillis((Integer) timeoutObj);
-            timeoutString = Utils.formatDurationAsTimespan(duration);
+        Long timeoutInMilliSec = getTimeoutInMilliSec(timeoutObj);
```

Comment:
The primary difference in behavior here is when a `timespan` is provided.

Previously, it simply returned back whatever had been provided as input.
Now, it first calls `parseTimeoutFromTimespanString(String inputAsTimespan)` to ensure it's a valid `timespan`, and then applies `calculateTimeoutPerServiceLogic(Long inputInMillis)` for consistent adjustment of its value.